### PR TITLE
Backport of docs: Terraform style guide into v1.1

### PR DESCRIPTION
**Pull request by teamterraform**
## Backport

This PR is auto-generated from #34876 to be assessed for backporting due to the inclusion of the label 1.1-backport.

:rotating_light:
>**Warning** automatic cherry-pick of commits failed. If the first commit failed,
you will see a blank no-op commit below. If at least one commit succeeded, you
will see the cherry-picked commits up to, _not including_, the commit where
the merge conflict occurred.

The person who merged in the original PR is:
@BrianMMcClain
This person should manually cherry-pick the original PR into a new backport PR,
and close this one when the manual backport PR is merged in.

> merge conflict error: POST https://api.github.com/repos/hashicorp/terraform/merges: 409 Merge conflict []

The below text is copied from the body of the original PR.

---

**Note:** This PR requires a redirect from the existing style conventions doc. Once this PR is merged, merge https://github.com/hashicorp/terraform-docs-common/pull/584

:mag: [Deploy preview](https://terraform-qqkzze3b9-hashicorp.vercel.app/terraform/language/style)

The Well-Architected Framework team has gathered feedback from HashiCorp employees and Terraform users in an attempt to build on our recommended style contentions. The intention of this guide is that it is used by those looking to form their own internal standardized style guide.

This guide is split into two parts:

1. Code style: Syntax conventions (brought from the existing style conventions document), resource naming and ordering, linting, variables, outputs, locals, etc.
2. Workflow style: Versioning, repo structure, module design, state management, secrets management, etc

This guide is a mix of prescriptive advice and non-prescriptive considerations, as some scenarios truly depend on the situation.

---

<details>
<summary> Overview of commits </summary>

  - ae1ccb8d4c54979c2ea6f6c8846811fea452b2e2  - be4434181a64144433d6f34814f2fcc843bdbdbb  - 55760eeb2ab660083308847904263c34281ee454  - 2a660d5663d534d6748f421a6eee4724ae2105b3  - dacaed0087feda430f2de113327cdf5494f51b3d  - eba067c8bc11c53fc6ec94c5ed587401ac168bf4  - 180165796e0067e39d6e7cf12945b225a987b21a  - df2c204a5b79831f63c857a1f11d418f24465c7b  - 6a9df4d6b2159ab462980e68efd290fcb441fef9  - 82a8c9e795b465bc6b36fddbc6179e8f5a7ed262  - 8c13af38565a0bb522ab4c73a8117b487505219c  - f39124b54e758943799f3f42799e412f5cf04504  - 40c34ef4b1a06467edb8a372239d22ea5f665298  - 4fcd7c0c4d017242b556332d4f4d4a647c8cca92  - 7e92d5f01c0af2f53ad9b6e690e2ffbcdc06dc1e  - 6f2f4a5f76b8d9547723d5ad2c3027ab81f0047a 

</details>

### CODE REVIEW

**Review by ritsok — APPROVED** [review]

File: website/docs/cli/commands/fmt.mdx [hunk 1]
```diff
@@ -9,7 +9,7 @@ description: >-
 
 The `terraform fmt` command is used to rewrite Terraform configuration files
 to a canonical format and style. This command applies a subset of
-the [Terraform language style conventions](/language/syntax/style),
+the [Terraform language style conventions](/terraform/language/style#syntax-conventions),
```
Comment (ritsok) [inline website/docs/cli/commands/fmt.mdx:12]:
```suggestion
the [Terraform language style conventions](/language/style#syntax-conventions),
```

File: website/docs/language/syntax/index.mdx [hunk 2]
```diff
@@ -18,6 +18,6 @@ section offer a more abstract view of the Terraform language.
   Terraform language. Terraform's JSON syntax is unfriendly to humans, but can
   be very useful when generating infrastructure as code with other systems that
   don't have a readily available HCL library.
-- [Style Conventions](/language/syntax/style) documents some commonly
+- [Style Conventions](/terraform/language/style#syntax-conventions) documents some commonly
```
Comment (ritsok) [inline website/docs/language/syntax/index.mdx:21]:
```suggestion
- [Style Conventions](/language/style#syntax-conventions) documents some commonly
```

**Review by radeksimko — COMMENTED** [review]
Nitpick: Would you mind cherry-picking this small fix https://github.com/hashicorp/terraform/pull/34934 here?

**Comment by BrianMMcClain** [comment]
Nice catch, thank you! After some discussion we actually decided we probably shouldn't backport this document this far, so I'm actually going to close this PR. I'll cherrypick this fix into the 1.5 backport PR as you mentioned, though!

**Pull request closed without merging**